### ISSUE TITLE
diesel migration run panicked

### ISSUE THREAD

**Issue by weiznich** [sgrif/diesel#140]
```
$ ls migrations/20160126232310_add_foo    
down.sql  up.sql  up.sql~
$ diesel migration run    
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: MigrationError(UnknownMigrationFormat("migrations/20160126232310_add_foo"))', ../src/libcore/result.rs:688
```
It seems like diesel migration run is panicking if there are additional files in the migration dir.
(Workaround: Delete those files).

**Comment by mfpiccolo** [comment]
This has been discussed here #64 and it was determined that only files with a leading `.` would be ignored.

**Comment by weiznich** [comment]
*~ are recovery files created by most editors. I think such files should be allowed/ignored. Furthermore there may be other examples(emacs uses #filename# for example).
I see three possibilities here: 
 1. maintain a list of "allowed" file endings
 2. simply check if there is a down.sql and a up.sql in the migrations dir
 3. remove all "temporary" files from the migrations dir

**Comment by mfpiccolo** [comment]
I don't like the idea of removing files from the directory.  The maintained list seems reasonable since we can add known files like the ones you mentioned and issues will be opened for other use-cases that come up.  Option 2 seems to be reasonable to me as well.

Were there other reasons aside from keeping these dirs clean for not allowing other files? @sgrif

**Comment by sgrif** [comment]
I think the most "correct" thing to do here is to ignore files that are ignored by VCS, but that feels a bit like overkill. I'm fine with just making sure `up.sql` and `down.sql` are there. We should also make sure that we better communicate our errors instead of just panicing

**Comment by sgrif** [comment]
Fixed by #141
